### DISCUSSION TITLE
주제 선정 아이데이션

### DISCUSSION THREAD

**Discussion by Jiisou** [MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone#4] [Ideas]
(https://impactlibrary.net/entry/UN-SDGs-info : UNsdgs 사회적 가치 관련 참고 자료)

사전에 깃헙을 활용해 각자의 아이디어를 수집 및 정리하고 이를 바탕으로 서로 의견을 주고 받고자 Discussion을 열었습니다.

- 서비스 방향성과 의도한 타겟층
- 해당 서비스의 사회적 영향력
- 주요 인공지능 기술
- 응용 가능할 것 같은 깃헙 오픈소스 (서비스 주기능 관련이되어도 좋고 평소에 활용해보고 싶었다 하는 것도 좋을 것 같아요)
정도의 요소를 필수적으로 포함한 간단한 기획을 부탁드리겠습니다

05-12 일까지 각자 페이지 작성해주시고 
05-14 일까지 다른 사람 아이디어에 하나 이상 피드백 댓글 달아주시면 좋을 것 같아요!
05-15 일 대면 회의에서 아이디어 수렴 및 제안서 완성, 깃헙 레포지토리 생성 진행 예정

**Comment by im3zero** [comment]
서비스 이름: **시니어층을 위한 음성인식 & 사물인식 앱**

- 음성인식 기능과 사물인식 기능이 있는 앱
    - 음성으로 질문하면 음성으로 대답하는 기능
        - 예) 시리, 빅스비 등의 대답기능
    - 음성인식 후 텍스트로 변환
        - 예) 클로버노트
        - 예) 변환된 텍스트를 바로 복사할 수 있는 기능(?)
- 사진을 찍거나 갤러리에 있는 사진을 인식 시키면 어떤 사물인지 알려주는 기능
- https://github.com/leehaesung/YOLO-Powered_Robot_Vision/blob/master/README_Korean.md
- 텐서플로우 api 사용 사물인지
    - https://nitr0.tistory.com/266
    - https://github.com/tensorflow/models

서비스 방향성: 생활 편의성 증대, 원활한 스마트폰 사용, 사회적 연결성 강화, 텍스트 입력에 익숙하지 않은 시니어들과 스마트폰 자체의 사용을 불편해하시는 시니어들의 원활한 스마프톤 사용

의도한 타겟층: 시니어층 (약 65세 이상)

해당 서비스의 사회적 영향력: 시니어층의 디지털 소외 디지털 소외 문제 해결

주요 인공지능 기술: 사물인식신경망, 자연어 처리, HMM, N-gram, 신경망, SD 

응용 가능할 것 같은 깃허브 오픈소스: 

https://github.com/rtzr/Awesome-Korean-Speech-Recognition?tab=readme-ov-file

→ 음성인식

https://github.com/dldudcks1779/YOLO-Object-Detection

https://github.com/leehaesung/YOLO-Powered_Robot_Vision/blob/master/README_Korean.md

https://github.com/tensorflow/models

→ 사물인식



<img width="413" alt="음성인식예시" src="https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/155755210/2b7f682d-a43a-4c59-99c1-dcbc3220b07a">

<음성인식예시>


<img width="433" alt="사물인식예시" src="https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/155755210/22fe7e0f-641d-4ea6-836d-8f6fc6fe0628">

<사물인식예시>

참고 자료

https://www.ibm.com/kr-ko/topics/speech-recognition

https://youtu.be/uqapSC9Spos?si=dNyf1doKkO5dJ_bK

https://www.aitimes.com/news/articleView.html?idxno=144027

**Reply by kwakmoonjung** [reply]
사물인식 관련 오픈소스로 웹캠과 라즈베리파이를 이용한 프로젝트를 제안해주셨습니다.
만약 라즈베리파이를 이용할 경우, 음성인식을 위해서는 음성 데이터를 수집하는 별도의 마이크가 필요할 것 같습니다.
혹시 관련하여 생각해두신 모델이 있으신지, 마이크와 라즈베리파이를 이용하여 음성인식을 진행한 오픈소스 자료가 있는 지 궁금합니다.

그리고 라즈베리파이를 이용할 경우, 핸드폰과 통신하는 방법도 궁금합니다. (블루투스/소켓통신/원격접속 등)

**Reply by GaeunLee1120** [reply]
사회적 영향력으로는 ‘시니어층의 디지털 소외 문제를 해결할 수 있다.’ 로 가면 어떨까요? 주제 자체가 어르신들의 편의를 제공하기 위함이기 때문에 서비스 자체의 이점은 충분하다고 생각합니다!

글을 읽고, 어떤 시나리오로 구동이 되는 지 좀 더 설명이 있었으면 좋겠다는 생각이 들었어요! 그리고 스마트폰 사용을 쉽게 하니까 어플 개발로 생각하시는 것 같은데 이와 관련해서도 간단한 의견 남겨주시면 좋을 것 같습니다. 

예제를 사진으로 남겨주시니 이해가 쉬워서 좋아요! 수고하셨습니다 😉

**Comment by Jiisou** [comment]
- 타겟층 : 독거 노인
   
65세 이상의 홀로 사는 노인을 지칭.
   가족, 친구, 이웃 등 **사회적 관계망과의 교류가 단절**되고 **사회적 역할 상실**에 따른 외로움과 고립감 등으로 **사회 단절에 따른 문제와 심각성**이 매우 높다.
    
- 기대효과 : 사회적 연결성 강화, 가속화 되고 있는 고령화 사회/시대에 대응

- 주요 사항
    - 구현 방법 : 앱
        - ui 오픈소스 
           https://github.com/mui/material-ui
    - 음성을 인식하여 화면 전환 - 디지털 사용 난이도를 쉽게 하기 위함
        - https://github.com/rtzr/Awesome-Korean-Speech-Recognition
    - 표정에서 감정 인식 → 감정 분류에 따라 적절한 음성 출력 - 고립감 및 단절 해소
        - https://github.com/petercunha/Emotion
    - 사용자가 시간대 별로 사진을 찍어서 사물 인식 - 건강한 루틴 형성 (뭔가.. 기술과 사용하는 상황 간의 연결성이 빈약한? 끼워맞춘..느낌)
    
노인 돌봄을 위한 제안된 ai 서비스 → 1인 가구임에 보다 중점을 두고 구체화할 수 있을 것 같다.

- 제안 기능 (전체)
    - 기상하여 특정 장소의 사진을 찍으면(객체 인식) “좋은 아침” 같은 응원 목소리
        - 아니면 웃는 얼굴 사진 기록
            - 귀여운 것 같아서 ㅎㅎ.. 매일 아침 웃는 사진 찍도록, 인증 완료되면→ “웃는 얼굴이 어쩌구 오늘 하루도~!@”
             (참고자료 https://mindcompass.github.io/multicampus/project3_3_emotion_detector/)

        - 기준 시간보다 늦어지면 이상 감지 (어딘가에 연락 가게,, 근데 어디로? 지자체??
        
    - 설정 시간마다 약물 복용 안내 목소리
    - 하루 한 번 안부 확인 전화(대화) → 말 동무
        - 대화의 수준? 자칫 넘 복잡해질 수 있겠다?
        
    - 유저층이 확대되면 사용 위치 기반으로 친구 매칭

- 당위성
    '서울 독거노인 35만명 넘어... 10명 중 3명 "고독사 우려"', 연합뉴스
    '당신의 미래도 독거노인...노인 3명 중 한 명 '독거'', SBS Biz
    한국보건72호 : [독거노인의 생활실태 및 정책과제] (https://repository.kihasa.re.kr/bitstream/201002/5962/1/72%ED%98%B8-%EC%B5%9C%EC%A2%85.pdf)

**Reply by kwakmoonjung** [reply]
독거노인을 위한 다양한 기능을 제안해주셔서, 실제로 구현되면 정말 좋을 것 같다는 생각이 됩니다.
그러나 이 모든 기능을 구현하기에는 시간적 여유가 없을 것 같아서, AI 기술을 활용한 ‘장소를 인식하여 응원 음성 출력’ 및 ‘웃는 얼굴 인식 후 응원 음성 출력’ 기능 위주로 구현하면 좋을 것 같습니다.

그리고 궁금한 점이 두 가지 있습니다.

제안해주신 기능에서, 특정 장소의 사진을 찍으면 응원 목소리가 나온다고 하셨는데, 여기서 정확히 무엇을 인식하고자 하는 것인지 궁금합니다.

1. 여러 객체를 인식하여 장소를 인식, 관련 음성 출력
ex) 숟가락, 그릇, 식탁 등의 요소를 인식하여 ‘주방’임을 인식
→ ‘맛있는 식사 하세요’ 음성 출력

2. 하나의 객체를 인식하여 관련 음성 출력
ex) TV 인식 시 드라마 추천, 창문 인식 시 날씨 언급 등

**Reply by GaeunLee1120** [reply]
제가 제안한 서비스와 비슷한 점이 많아 놀랐습니다 ㅎㅎ ~~ 저도 독거 노인의 안부를 묻는 서비스를 제안했는데 전반적인 process도 비슷한 것 같습니다.

주제를 보면 사회적 영향력 & 문제 해결 부문에서도 장점이 확실한 것 같아요! 그리고 공통적으로 제가 코멘트를 남기는데, 관련 기술과 개발 방식 (웹, 어플, 등등), 시나리오 등을 적어주시면 이해가 더 잘 될 것 같습니다. 

작성 수고하셨습니다 !! 😘

**Comment by kwakmoonjung** [comment]
![image](https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/161675231/d78b6497-4681-43bf-9f7f-0d74500bf50a)
‘인공지능 기술 기반’ 

→ 그나마 관련 지식이 있어서, 컴퓨터 비전을 활용하고 싶습니다!

## 주제

**“시각 장애인을 위한 물체 인식 시스템”**


### 주요 기능

**컴퓨터 비전을 통한 장애물 인식**, 이에 따른 간단한 경로 생성, 장애물 및 경로를 **음성으로 출력**

ex) 앞에 나무가 있을 경우, ‘앞에 나무가 있습니다. 왼쪽으로 피하십시오’ 음성 출력


### 서비스 방향성과 사회적 영향력

→ ‘시각 장애인 보행’을 보조하는 시스템 구축 필요!!

1. **최근 공유형 전동킥보드로 인한 시각장애인의 위험성 증가**
- 킥보드는 부딪히는 걸로 끝나지 않고 넘어지는 게 문제
    
    [[점자블록 위 불쑥…방치된 킥보드, 시각장애인에겐 ‘지뢰’](https://www.hani.co.kr/arti/society/society_general/968605.html)](https://www.hani.co.kr/arti/society/society_general/968605.html)
    

 

- 2022년 서울시 접수된 전동킥보드 관련 민원 총 1,194건. 이 중 무단방치와 견인 관련 민원이 86%
    
    [[시각장애인의 생명선…"점자블록 위 킥보드 NO"](https://news.cpbc.co.kr/article/1109628)](https://news.cpbc.co.kr/article/1109628)
    

1. **시각 장애인을 위한 보행 보조 시스템 부족**
- 16년 교통약자 이동 편의 실태조사에 따르면 시각장애인의 보도 이용 만족도는 비장애인 평균 58점에 비해 낮은 46점으로 나타남.
    
![image](https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/161675231/9f98e61f-1131-42d1-970c-2f36efd1123a)
    

### 관련 오픈소스

- 예제 1 : ESP-32 카메라 모듈 및 opencv, yolov3를 이용한 객체 인식
    
    [[ESP32-Cam과 OpenCV, Yolo를 이용한 객체검출](https://blog.naver.com/jcosmoss/222976730180)](https://blog.naver.com/jcosmoss/222976730180)
    
    → [[yolov3의 cfg, weight파일](https://pjreddie.com/darknet/yolo/)](https://pjreddie.com/darknet/yolo/)과 분류할 [[class 파일을 다운](https://github.com/pjreddie/darknet/blob/master/data/coco.names)](https://github.com/pjreddie/darknet/blob/master/data/coco.names)받아옴
    
    → 블로그에 있는 코드 실행
    

- 예제 2 : IP카메라와 yolov3 / yolov5 을 이용하여 객체 인식
    
    (두가지 yolo 버전을 사용한 설명이 모두 나와 있음)
    
    [[6. Yolo를 이용한 실시간 영상 객체 감지!! Yolo v5](https://naver.me/GI86nZDS)](https://naver.me/GI86nZDS)
    
    1. 카메라 데이터 수신
        - 카메라 기종에 따라 opencv에서 이미지를 처리하지 못하는 경우가 있음.
        - 데이터 전처리를 통해 수신 데이터를 실시간으로 파이썬에서 처리할 수 있게 함
    2. [[yolo 홈페이지](https://naver.me/Fh6MxUXt)](https://naver.me/Fh6MxUXt)에서 기존에 학습된 cfg, weight파일 및 분류할 수 있는 데이터셋을 가져옴
        - yolov5을 사용할 경우, 추가적인 개발 환경 세팅 필요
            
            → Cuda, Anaconda, Pytorch 설치 필요
            
    3. 블로그에 나와 있는 코드 실행
        - 전체 소스코드
            
            [[GitHub - Tory-Hwang/Object_Detect](https://naver.me/5wABdEly)](https://naver.me/5wABdEly)
            

### 참고자료

[[시각장애인의 보행을 위한 경로 탐색 및 횡단보도, 신호등 검출 시스템 구현.pdf](https://prod-files-secure.s3.us-west-2.amazonaws.com/6989b968-748f-4b18-a005-9c616ea829cd/bdbf32e5-6a80-4666-b9d8-8db6a328d54c/%EC%8B%9C%EA%B0%81%EC%9E%A5%EC%95%A0%EC%9D%B8%EC%9D%98_%EB%B3%B4%ED%96%89%EC%9D%84_%EC%9C%84%ED%95%9C_%EA%B2%BD%EB%A1%9C_%ED%83%90%EC%83%89_%EB%B0%8F_%ED%9A%A1%EB%8B%A8%EB%B3%B4%EB%8F%84_%EC%8B%A0%ED%98%B8%EB%93%B1_%EA%B2%80%EC%B6%9C_%EC%8B%9C%EC%8A%A4%ED%85%9C_%EA%B5%AC%ED%98%84.pdf)](https://prod-files-secure.s3.us-west-2.amazonaws.com/6989b968-748f-4b18-a005-9c616ea829cd/bdbf32e5-6a80-4666-b9d8-8db6a328d54c/%EC%8B%9C%EA%B0%81%EC%9E%A5%EC%95%A0%EC%9D%B8%EC%9D%98_%EB%B3%B4%ED%96%89%EC%9D%84_%EC%9C%84%ED%95%9C_%EA%B2%BD%EB%A1%9C_%ED%83%90%EC%83%89_%EB%B0%8F_%ED%9A%A1%EB%8B%A8%EB%B3%B4%EB%8F%84_%EC%8B%A0%ED%98%B8%EB%93%B1_%EA%B2%80%EC%B6%9C_%EC%8B%9C%EC%8A%A4%ED%85%9C_%EA%B5%AC%ED%98%84.pdf)

: yolov8로 객체 인식, 인식한 객체와 보행 경로를 애플리케이션에서 음성으로 안내

- (참고) 다른 주제 : AI의 ‘데이터 편향성’ 문제
    
    데이터 편향성 (Data Bias) : 학습 데이터가 편향되어, AI 결과물 또한 편향 문제를 보이는 것
    

    **예시 1)** AI가 특정 인종/성별에 대한 차별
    
    - Stable Diffusion(이미지 생성 AI)에 'unprofessional한 사람의 이미지' 요청 시 '고령의 흑인 남성'과 같은 특정 인종과 성별을 반영한 이미지를 그림
    ![image](https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/161675231/94d4073d-9e08-40df-9492-67727369298e)
    
    **예시 2)** AI마다 정치 및 경제 문제에 대해 다른 경향성을 보임
    
    - 챗GPT나 LLaMA와 같은 초거대언어모델(LLM)에 정치 및 경제 문제에 대한 62가지의 질문을 던지고 ‘긍정’ 혹은 ‘부정’으로 답하도록 지시함
        
        → 챗GPT는 ‘진보’적인 성향, LLaMA의 답변은 비교적 ‘보수’적인 경향
        
    
    **예시 3)** 글로벌 기업은 데이터 다양성을 중요시함
      ![image](https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/161675231/67d27550-a683-4246-89c6-db9306737579)
    
    **해결 방안)** 
    
    *(각각 자세히 찾아보면 논문 및 알고리즘이 있으나, 우리가 사용하기에는 너무 수준이 높은 듯합니다,,,)* 
    
    1. 전처리를 통해 고른 분포의 학습 데이터 만들기
        - 데이터 전처리를 통해 데이터가 특정 그룹이나 특성에 치우치는 경향을 줄일 수 있음
    2. 다양한 학습 데이터 수집
    3. 학습 데이터 품질 평가
        - CV-HAZOP(CV Hazard and Operability Study) 체크리스트를 사용하여 데이터셋을 분석 및 평가하는 논문 존재
    4. AI모델 평가 및 감독
        - 인간의 개입이 필요한 평가 체계나 보정 메커니즘을 도입
    
    **참고 자료)**
    
    [[생성형 AI와 데이터 편향성 문제: 데이터 편향이란 무엇이며 어떻게 해결할 수 있을까?](https://blog-ko.superb-ai.com/generative-ai-and-the-data-bias-problem-what-is-data-bias-and-how-can-it-be-solved/)](https://blog-ko.superb-ai.com/generative-ai-and-the-data-bias-problem-what-is-data-bias-and-how-can-it-be-solved/)

**Reply by GaeunLee1120** [reply]
우선 주제가 흔치 않은 주제라, 서비스를 개발하게 된다면 큰 장점이 될 것 같습니다!
크게 ‘시각장애인을 위한 물체 인식 시스템’과 AI의 ‘데이터 편향성’ 주제를 제안해주신 것 같은데 
궁금한 점은, 이러한 서비스를 어떻게 사용할 수 있는지 개발 방법이 적혀있지 않은 것 같아서요!
(웹, 어플, 단순 pc용 등등) 관련해서 간단하게라도 구현 방식을 적어주시면 의도를 파악하는데 조금 더 도움이 될 것 같습니다 👍👍

**Reply by kwakmoonjung** [reply]
가은님 답변)
아무래도 보행을 위한 서비스이니만큼, 앱으로 구현하는 게 사용자가 일상에서 쉽게 사용할 수 있을 것 같습니다.
좋은 피드백 감사합니다!

(데이터 편향성의 주제는 웹이나 앱으로 구현할 필요성까지는 없을 것 같습니다. 단순하게 기술만 구현하기에도 어려울 것 같아서, 웹이나 앱까지 개발할 시간이 부족할 것 같아요)

**Reply by Jiisou** [reply]
오옷 다양한 관점에서 색다른 주제를 제안해주셨군요! 흔치 않은 적용이었을텐데 프로젝트에 활용할 수 있는 방법에 대해 많이 고민해보시고 정리해주신 노고가 느껴지는 듯 합니다,, 
오토인코더나 GAN 같은 유명한 생성모델 소스를 가지고 다양성을 보장하는 학습 데이터 확보를 위해 ai를 활용해볼 수도 있을 것 같아요(?)
tmi지만 지난 학기에 융사글 과목을 수강하며 해당 주제에 대한 소논문을 작성한 경험이 있어서 괜히 반가웠습니다 ㅎㅎ.

**Reply by im3zero** [reply]
가은님 말처럼 흔하지 않은 주제라고 생각해 잘 활용하면 너무 좋을 것 같습니다!!
시각 장애인의 경우 흰지팡이를 주로 사용하는데 지팡이와 사용되는 용도가 거의 비슷한 것 같아서 살짝쿵 차별화되는 기능이 있으면 더 더 좋을 것 같다는 생각이 들었습니다!
너무 수고하셨어요 😆😆!!

**Comment by GaeunLee1120** [comment]
# 독거 노인(1인 가구)의 고독사 문제 해결을 위한 서비스 제안

### 문제

- **독거 노인(1인 가구)의 고독사**
    - 1인 가구가 늘어나며 안부를 확인하는 서비스는 다양한 형태로 존재하지만, 방문 시에도 구체적인 위치, 경위 등을 몰라 긴급한 대처가 어려운 상황이 있다. 이를 위해 **Vision**과 **음성 인식 기술**을 이용하면 정확한 위치나 시간, 경위 등을 파악해 보다 신속한 대처가 가능하다.
 ![image](https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/138747086/00dcb10e-9a5e-4a56-839c-b95eb100b637)
      - [관련 뉴스 기사](https://n.news.naver.com/article/025/0003358753?sid=101)

  - [독거노인 고독사 문제 & 비율]
  ![image](https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/138747086/5e99536d-a40c-4692-80f5-300bf409ab9e)



### 비슷한 서비스 예시

- **AI 스피커**
  - [관련 뉴스 기사](https://n.news.naver.com/article/417/0001001592?sid=101)
- **안심앱 (일정 시간 미사용 시 알림)**
  - [관련 뉴스 기사](https://n.news.naver.com/article/021/0002634901?sid=102)

### 제안하는 서비스

**Vision** 및 **음성 인식 기술**을 이용한 독거 노인(1인 가구) 안부 확인 서비스

- **아이디어**: 매일유업의 독거 노인 안부 확인 우유 배달 서비스에서 차용
- **목표**: 독거 노인(1인 가구)의 안부 확인 서비스 제공

#### 서비스 프로세스

1. 매일 아침 인삿말을 건네고, 대답으로 기상 여부 확인 (음성 인식 기술 개발 및 알고리즘 설계)
2. 일정 시간이 지나도 응답이 없을 경우, 2차 확인을 위해 카메라로 상황을 확인하도록 알람 (사용자 얼굴 인식을 위한 학습된 모델 개발 및 알고리즘 설계)
3. 기준 이상의 시간 경과 시 비상 연락망으로 연결 (구동 시나리오 작성 및 로그 생성: DB)

- AI 스피커와 차별화: 어르신들은 디지털 기기를 다루는데 어려움을 겪는 비율이 높으므로, 접근성이 좋은 스마트폰의 음성 인식 서비스 또는 데스크탑 PC 전용 서비스를 이용
- 시나리오: "좋은 아침입니다, 오늘 컨디션은 어떠신가요?" 등의 간단한 인삿말로 안부 확인. 응답이 없으면 카메라로 상황 확인, 일정 시간 동안 얼굴이나 모션이 인지되지 않으면 가족이나 비상 연락망으로 연락

#### 기술 세부 구현 및 GitHub 레포지토리

- **음성 인식 기술 개발 및 알고리즘 설계**
  - **TTS (Text to Speech)**: 글자를 음성으로 변환 (묻는 인삿말). 별도의 모델 필요 X, 시나리오 상 문장을 입력만 해 놓으면 AI 화자를 이용해 구현 가능
  - **STT (Speech to Text)**: 음성을 인식하여 문장으로 생성 후 판단 (사용자의 음성을 수집하기 위한 목적). 
    - **Kospeech**: 한국어만 지원하는 모델, PyTorch 기반
      - [Kospeech GitHub 레포지토리](https://github.com/sooftware/kospeech#introduction)
    
- **사용자 얼굴 인식을 위한 학습된 모델 개발**
  - **Labeled Faces in the Wild**: 99.38%의 정확도를 자랑하는 데이터셋 LFW
  - OpenCV와 함께 사용
    - [face_recognition GitHub 레포지토리](https://github.com/ageitgey/face_recognition/blob/master/README_Korean.md)
    - [unknown_face_classifier GitHub 레포지토리](https://github.com/ukayzm/opencv/tree/master/unknown_face_classifier)

- **로그 생성: DB**
  - **MySQL**: 단순 로컬 PC 사용 시 굳이 서버 필요 X
  - 안드로이드도 MySQL 연동 가능하나 많은 로그 필요 없을 경우 서버 구축 필요 X
  - 단순히 몇 개의 시나리오만 저장해 UI 상으로 보여주는 것만 구현

### Reference

- **아이디어 출처**: 매일유업 우유 배달 - 안부 확인 사업
  - [착한우유 사업](https://www.chakanmilk.com/m/index.do)

**Reply by kwakmoonjung** [reply]
관련 자료를 자세하고 다양하게 조사하신 것 같습니다. 또한 서비스의 시나리오를 구체적으로 적어주셔서 알고리즘 흐름이 어떻게 구성될 지 이해하기 쉬웠습니다. 준비하시느라 고생하셨어요 :)

가은님 내용과 관련하여 질문이 세 가지가 있습니다.

**질문 1. 2차 확인의 절차가 다음과 같은 흐름이 맞을까요?**
1. PC가 카메라를 작동하여 주변 환경을 파악
2. 알람을 작동하여 사용자가 카메라 근처에 오게 한 후, 얼굴인식을 진행
3. 사용자가 감지되지 않을 경우, 비상 연락망으로 연락

**질문 2. ‘어르신의 움직임을 확인’한다는 것이 다음의 세 가지를 내포하고 있는 게 맞는지 궁금합니다.**
1. 사람이 있는지 인지
2. 사람이 움직이는지 인지 → bounding box 속도를 통해 파악?
3. 사람이 사용자(=어르신)이 맞는지 인지 → 사용자가 아닐 경우에도 비상 연락망으로 연락?

**질문 3. 사용자 얼굴 인식 모델인 LFW의 경우**에는, 학습 데이터에 있는 사람의 얼굴만 인식할 수 있는 것으로 이해했습니다. 그렇다면 이 학습 데이터에 있는 노인의 이미지를 찾아서 사용하실 건가요?

또는 아래 첨부해주신 unknown_face_classifier 자료처럼 모르는 얼굴을 실시간으로 구분하는 기능을 구현할 것인지 궁금합니다.

그리고 unknown_face_classfier 예제에서 한 사람의 얼굴을 저장해두고 이 얼굴과 비슷한지/다른지 구분할 수도 있는지 궁금합니다.

**Reply by Jiisou** [reply]
사전 기획이라는 말이 무색할 정도로 정리가 너무 잘되어있어 감탄했습니다…
서비스가 특화된 시나리오에서 음성 인식→컴퓨터 비전의 연결이 특히 넘 잘 짜여져 있는 것 같아요!!

**Reply by GaeunLee1120** [reply]
> 관련 자료를 자세하고 다양하게 조사하신 것 같습니다. 또한 서비스의 시나리오를 구체적으로 적어주셔서 알고리즘 흐름이 어떻게 구성될 지 이해하기 쉬웠습니다. 준비하시느라 고생하셨어요 :)
> 
> 가은님 내용과 관련하여 질문이 세 가지가 있습니다.
> 
> **질문 1. 2차 확인의 절차가 다음과 같은 흐름이 맞을까요?**
> 
> 1. PC가 카메라를 작동하여 주변 환경을 파악
> 2. 알람을 작동하여 사용자가 카메라 근처에 오게 한 후, 얼굴인식을 진행
> 3. 사용자가 감지되지 않을 경우, 비상 연락망으로 연락
> 
> **질문 2. ‘어르신의 움직임을 확인’한다는 것이 다음의 세 가지를 내포하고 있는 게 맞는지 궁금합니다.**
> 
> 1. 사람이 있는지 인지
> 2. 사람이 움직이는지 인지 → bounding box 속도를 통해 파악?
> 3. 사람이 사용자(=어르신)이 맞는지 인지 → 사용자가 아닐 경우에도 비상 연락망으로 연락?
> 
> **질문 3. 사용자 얼굴 인식 모델인 LFW의 경우**에는, 학습 데이터에 있는 사람의 얼굴만 인식할 수 있는 것으로 이해했습니다. 그렇다면 이 학습 데이터에 있는 노인의 이미지를 찾아서 사용하실 건가요?
> 
> 또는 아래 첨부해주신 unknown_face_classifier 자료처럼 모르는 얼굴을 실시간으로 구분하는 기능을 구현할 것인지 궁금합니다.
> 
> 그리고 unknown_face_classfier 예제에서 한 사람의 얼굴을 저장해두고 이 얼굴과 비슷한지/다른지 구분할 수도 있는지 궁금합니다.

피드백 감사합니다! 질문해주신 것에 답변할게요~
질문1 : 사용자 얼굴 또는 모션 인식 후 긴급상황이라고 판단될 시 알람이 가는 과정이 맞습니다. 
질문2 : 제가 생각하기에는 일정 속도 혹은 bounding box의 위치 값 변동 이상이 있으면 움직임이 있다고 설계하려고 했습니다.
질문3 : 노인의 데이터만 학습시키는 것보다는 다양한 연령대 혹은 외형의 이미지를 학습시키는 것이 좋다고 생각합니다.  개인 맞춤 기능을 제공한다면 말씀해주신 예제(unknown_face_classifier)가 적절할 거라고 생각하지만, 이것은 좀 더 팀원끼리 develop 해보면 좋을 것 같아요

**Reply by im3zero** [reply]
저도 시나리오를 보고 이해가 쉬웠습니다! 구체적인 설명 너무 감사해요!! 
(단순한 궁금증에 ..) pc 서비스를 제안해주셨는데, 가정에 pc가 존재하지 않는 경우는 어떻게 하는 것이 좋을까요?

**Reply by GaeunLee1120** [reply]
> 저도 시나리오를 보고 이해가 쉬웠습니다! 구체적인 설명 너무 감사해요!! (단순한 궁금증에 ..) pc 서비스를 제안해주셨는데, 가정에 pc가 존재하지 않는 경우는 어떻게 하는 것이 좋을까요?

일단, 일반적으로 가정에 PC가 있다는 가정 하에 기획을 했던 거라 PC가 없는 가정에서는 사용할 수 없는 서비스가 될 것 같아요! 이 경우는 제안서나 최종적으로 정리할 때 develop 할 방향으로 제시하면 좋을 것 같습니다!
